### PR TITLE
C++ array writer

### LINKED ISSUE #134: Implement C++ arrays
Implement C++ code gen for arrays

### LINKED ISSUE #135: Test C++ arrays
Test C++ code gen for arrays

### PR DESCRIPTION
**Pull request by tiffany1618**
Closes #134
Closes #135 

Added:
- `ArrayCppWriter`: writes C++ for FPP array definitions
- `StringCppWriter`: writes C++ classes for strings of given sizes
- `FormatCppWriter`: writes C++ format strings for FPP formats
- `TypeUtils`: adds some type utilities
- `CppWriterState.writeIncludeDirectives()`: writes C++ include directives for used symbols
- Unit tests for C++ array writer

### CODE REVIEW

**Comment by Joshua-Anderson** [comment]
@tiffany1618 Disregard my comment, this looks great! I saw the implementation of operators that the string base implemented and didn't appreciate that those were the implementation for the array, not the implementation for the string class.

File: compiler/tools/fpp-to-cpp/test/array/PrimitiveU16ArrayAc.ref.cpp [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      Serializable()
+  {
+    for (U32 index = 0; index < SIZE; index++) {
+      this->elements[index] = e;
+    }
+  }
+
+  PrimitiveU16 ::
+    PrimitiveU16(
+        const ElementType (&e1),
```
Comment (bocchino) [inline compiler/tools/fpp-to-cpp/test/array/PrimitiveU16ArrayAc.ref.cpp:54]: The parentheses around (&e1) are unusual. Can we write `const ElementType& e1` as elsewhere?

**Review by bocchino — COMMENTED** [review]
Looks great! Please see the comment.

File: compiler/tools/fpp-to-cpp/test/array/PrimitiveU16ArrayAc.ref.cpp (hunk 1 above)
Comment (tiffany1618) [reply compiler/tools/fpp-to-cpp/test/array/PrimitiveU16ArrayAc.ref.cpp]: Ok, I changed it and updated the unit tests.

Comment (bocchino) [reply compiler/tools/fpp-to-cpp/test/array/PrimitiveU16ArrayAc.ref.cpp]: Looks good!

**Pull request merged**